ci/test/lint-deps.toml

```diff
@@ -0,0 +1,69 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Configuration file for cargo-deplint
+#
+# If you find yourself here because CI failed on a PR, please don't blindly edit
+# out lints :D. Danhhz is always happy to discuss crate structure if you need
+# help fitting a change into these rules.
+
+# Keep various large crates from depending on frequently changing ones, so that
+# development iteration doesn't invalidate them.
+[[deny]]
+name = "mz-transform"
+dependencies = [
+    "mz-compute-client",
+    "mz-storage-client",
+]
+
+[[deny]]
+name = "mz-sql"
+dependencies = [
+    "mz-compute-client",
+    "mz-storage-client",
+]
+
+[[deny]]
+name = "mz-compute"
+dependencies = [
+    # TODO: "mz-storage-controller",
+    "mz-storage",
+]
+
+[[deny]]
+name = "mz-storage"
+dependencies = [
+    # TODO: "mz-storage-controller",
+    "mz-compute",
+]
+
+# Keep various crates in only one of environmentd or clusterd.
+[[deny]]
+name = "mz-environmentd"
+dependencies = [
+    "mz-compute",
+    "mz-rocksdb",
+    "mz-storage",
+    "mz-storage-operators",
+]
+
+[[deny]]
+name = "mz-clusterd"
+dependencies = [
+    "mz-storage-client",
+    # TODO: "mz-storage-controller",
+]
+
+# Persist is meant to be a strong enough abstraction that it doesn't depend on
+# mz internals like Row.
+[[deny]]
+name = "mz-persist-client"
+dependencies = [
+    "mz-repr",
+]
```

Comment:
This is totally fine, but it might be nice for `cargo-deplint` to support embedding denies directly in each crate’s Cargo.toml. Cargo has a blessed way of allowing this sort of custom metadata in the Cargo.toml.

Reply:
Hmmm that's interesting, but I think there's decent value in having them all in one place. Makes it easier to reason about the graph as a whole. I wonder if there's some happy middle ground...